src/pages/home/report/EmojiPickerMenu/index.js

```diff
@@ -44,12 +48,38 @@ class EmojiPickerMenu extends Component {
         // If more emojis are ever added to emojis.js this will need to be updated or things will break
         this.unfilteredHeaderIndices = [0, 33, 59, 87, 98, 120, 147];
 
+        // Toggles which keys the search input will listen to
+        // NOTE: these need to be instance members so we can
+        // reference the same event handlers in memory when removing them
+        this.keyToDefaultPreventer = {};
+        ['ArrowDown', 'ArrowLeft', 'ArrowRight', 'ArrowUp'].forEach((key) => {
+            this.keyToDefaultPreventer[key] = (keyBoardEvent) => {
+                if (keyBoardEvent.key === key) {
```

Comment:
I've defined them in an object so that we can reference the same event handler for each arrow key. That way we can always be adding and removing the same event handlers in memory. We don't need to check if the key is in the array since `toggleKeysOnSearchInput` is only ever called with arrow keys.

Reply:
Doing it this way we're adding 4 event handlers to the `keyDown` event though so every time a key is pressed we run through the same method 4 times. I think a better way to do it would be to just have one method that is connected to `keyDown` and then keep the active keys in the state rather than connecting and disconnecting keyboard events. Something like this
```
onKeyPressed(key): {
    if (this.state.keysToAccept.contains(keyBoardEvent.key) {
        keyBoardEvent.preventDefault();
    }
}
```

And then instead of `this.toggleKeysOnSearchInput` as a method, you can just set the state to include the relevant keys [here](https://github.com/Expensify/Expensify.cash/pull/2532/files#diff-0aeb1e7eb3f2813b84c81ee9aaf16b4e67307297dbc129803f5f0eb20a6a3693R272-R276).

Reply:
Gootcha, thanks for clarifying what you meant!